ui/src/lib/yjs/useYNode.ts

```diff
@@ -38,11 +39,101 @@ export default ({
               return [...snids, newNode.id];
             });
           }
+
+          // For routers without routingPort, generate unique port name
+          const isRouterInput = newNode.data.outputs?.length;
+          const isRouterOutput = newNode.data.inputs?.length;
+
+          if (
+            (isRouterInput || isRouterOutput) &&
+            !newNode.data.params?.routingPort
+          ) {
```

Comment:
I am not sure this is sufficient for knowing whether we are dealing with an OutputRouter or InputRouter, is it? Won't most nodes being added meet these conditions with at least outputs if not inputs? 🤔 
If so, that means this check is pointless